instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -690,19 +690,23 @@ async def otel_send(message: dict[str, Any]):
                         duration_attrs[
                             SpanAttributes.HTTP_STATUS_CODE
                         ] = status_code
-                        set_status_code(server_span, status_code)
                         set_status_code(send_span, status_code)
 
                         expecting_trailers = message.get("trailers", False)
                     elif message["type"] == "websocket.send":
-                        set_status_code(server_span, 200)
                         set_status_code(send_span, 200)
                     send_span.set_attribute("type", message["type"])
-                    if (
-                        server_span.is_recording()
-                        and server_span.kind == trace.SpanKind.SERVER
-                        and "headers" in message
+
+                if (
+                    server_span.is_recording()
+                    and server_span.kind == trace.SpanKind.SERVER
+                ):
+                    if (message["type"] == "http.response.start") or (
+                        message["type"] == "websocket.send"
```

Comment:
Can you include `"websocket.http.response.start"`, please?

It's one of the ASGI extensions, and it's supported by uvicorn:
https://asgi.readthedocs.io/en/latest/extensions.html#websocket-denial-response

Reply:
It would be a different change/fix and hence belongs in a separate PR, IMO. The PR attempts to make the status codes, which were already set, independent of whether the inner spans are recorded or not.

Given that the PR has been open for almost a year, I realize that the PR does not seem to have much chance of getting merged. I'll close it to clean up the stale PR.